Xuan p patch 1

xuanP opened the pull request:
Bonjour Monsieur,
I have finished my part and it can do segmentation now. In case that you cannot run MATLAB on your computer, I took a screenshot of my final results.
New upload:
*main.m for running directly
*LinearSystem.m for output of article(11)
*TDMA.m used in LinearSystem to solve the diagonally dominant tridiagonal linear system
*Some small functions: 
      Init_square.m for an initial contour of the image.
      GetDifference.m just for getting the |gradient u|

New modify:
*stopFunction.m Correct the obsolete conv2() expression
*AOS.m Modify the iteration expression for levelset after adding a BalloonForce
*Upwind.m Correct the function name Dx_for etc. into lowercase dx_for etc. because the file name of matlab may go wrong under some versions.
Bien cordialement,
Xuan PENG

xuanP committed:
Update Upwind.m

xuanP committed:
Update AOS.m

xuanP committed:
Update stopFunction.m

xuanP committed:
Update readImage.m

xuanP committed:
Update AOS.m

xuanP committed:
LinearSystem.m

This file is used to get the upper diag and lower diag as well as solve the diagonally linear system equations

xuanP committed:
GetDifference.m

The function contained in this file is used in AOS.m for getting the |gradient u|(u is the levelset) used in the non erosion part

xuanP committed:
TDMA.m

This function used THOMAS Algorithm to solve the strictly diagonal matrix.A*x=R.
It will output x

xuanP committed:
main.m

This is the main program that you can run directly in matlab.
Other files are just functions

xuanP committed:
Ini_square.m

This function allows to draw an initial levelset contour( looks like) according to the image size. And the levelset contour will be used in the first iteration.

xuanP committed:
Update main.m

xuanP committed:
A Screenshot for showing results

Bonjour Monsieur,
In case that you cannot run MATLAB on your computer, I took a screenshot to show the result.

troussil commented:
Nice job. Thanks. 

Please, merge the master branch into your branch and resolve conflicts. Then I could merge this pull request. 

In addition, 
- think about coding it in another language (python, java, c...)  
- think about test cases
- compare different models (only balloon force, only diffusion term, geometric model, geodesic model...)

xuanP committed:
Update main.m

Add some lines in order to modify all parameters more easily

xuanP committed:
Update stopFunction.m

Change the expression of g: modify (k*) to (/k.^2). k is lambda in the article.

xuanP committed:
important update LinearSystem.m

Some errors happen when I wanted to use geometric model because of the using a as a variable but not an array.
So I changed a line of code.(from "a" to "a(:)")

xuanP committed:
Update AOS.m

Add two lines to test the geometric model more easily

xuanP committed:
Create readme

xuanP committed:
Results for 2 models

geometric&geodesic in 180 iterations; force const k=-0.3; sigma=1.2; lambda=1; timestep=5

xuanP commented:
Bonjour Monsieur,
On Tuesday's meeting, you wanted me to test Geometric model and draw a similar picture(with 4 items) as the picture tested in the article. There were some problems at that time and this new pull request solved them.
*Update LinearSystem.m in order to solve an inappropriate use of matlab's array.
*Update main.m in order to test different pictures using different parameters more easily.
*Update stopFunction.m: modify the expression of g(x).
*Update AOS.m: add two lines to test the geometric model
*Create a new folder containing the test results.

Question:
The Geometric model doesn't work very well and there exists some over-segmentation. Expect to discuss with you on Friday.

Thanks.
Xuan

troussil commented:
You must solve conflicts before I can merge your code into the main repo.
